virttest/vt_utils/cpu.py

```diff
@@ -0,0 +1,144 @@
+#
+# library for cpu related helper functions
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; specifically version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat (c) 2023 and Avocado contributors
+# Author: Houqi Zuo <hzuo@redhat.com>
+import re
+
+from avocado.utils import process
+
+
+def get_cpu_info():
+    """
+    Return information about the CPU information.
+
+    :return: cpu information
+    :rtype: list[dict]
+    """
+    cpu_info = list()
+    with open("/proc/cpuinfo") as fd:
+        processors = fd.read().strip().split("\n\n")
+    for processor in processors:
+        if processor != "":
+            info = dict(map(lambda x: [i.strip() for i in x.split(":", 1)],
+                            processor.split("\n")))
+            cpu_info.append(info)
+
+    return cpu_info
+
+
+def get_cpu_model_name():
+    """
+    Return physical cpu model.
+
+    :return: cpu model name
+    :rtype: string
+    :raises: An OSError will be raised if it's NOT available on the platform
+    """
+    cpu_model = process.run("lscpu").stdout_text
+    cpu_model_re = "(?m)^[mM]odel name:.*$"
+    cpu_model = re.search(cpu_model_re, cpu_model)
+    if not cpu_model:
+        raise OSError("The cpu model name was NOT found!")
+    cpu_model = cpu_model.groups()[0]
+    return cpu_model.split(":", 1)[-1].strip()
```

Comment:
Hi @nickzhq 
I hit an error when calling this function.
The following is the detail:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/yhong/kar/workspace/avocado-vt/virttest/vt_utils/cpu.py", line 57, in get_cpu_model_name
    cpu_model = cpu_model.groups()[0]
IndexError: tuple index out of range
```
Could help to check it?
Thanks in advance.

Reply:
> get_cpu_model_name

This bug will be fixed at patch#4030 ( https://github.com/avocado-framework/avocado-vt/pull/4030 ). Thanks!

Houqi